config_center/parser/configuration_parser.go

```diff
@@ -0,0 +1,250 @@
+			urlStr = urlStr + "&category="
+			urlStr = urlStr + constant.APP_DYNAMIC_CONFIGURATORS_CATEGORY
+			urlStr = urlStr + "&configVersion="
+			urlStr = urlStr + config.ConfigVersion
+			url, err := common.NewURL(context.Background(), urlStr)
+			if err != nil {
+				return nil, perrors.WithStack(err)
+			}
+			urls = append(urls, &url)
+		}
+	}
+	return urls, nil
+}
+
+func getServiceString(service string) (string, error) {
+	if len(service) == 0 {
+		return "", perrors.New("service field in configuration is null.")
+	}
+	var serviceStr string
+	i := strings.Index(service, "/")
+	if i > 0 {
+		serviceStr = serviceStr + "group="
+		serviceStr = serviceStr + service[0:i]
+		serviceStr = serviceStr + "&"
+		service = service[i+1:]
+	}
+	j := strings.Index(service, ":")
+	if j > 0 {
+		serviceStr = serviceStr + "version="
+		serviceStr = serviceStr + service[j+1:]
+		serviceStr = serviceStr + "&"
+		service = service[0:j]
+	}
+	serviceStr = service + "?" + serviceStr
+	return serviceStr, nil
+}
+
+func getParamString(item ConfigItem) (string, error) {
+	var retStr string
+	retStr = retStr + "category="
+	retStr = retStr + constant.DYNAMIC_CONFIGURATORS_CATEGORY
+	if len(item.Side) > 0 {
+		retStr = retStr + "&side="
+		retStr = retStr + item.Side
+	}
+	params := item.Parameters
+	if len(params) <= 0 {
+		return "", perrors.New("Invalid configurator rule, please specify at least one parameter " +
+			"you want to change in the rule.")
+	}
+	for k, v := range params {
+		retStr = retStr + "&"
+		retStr = retStr + k
+		retStr = retStr + "="
+		retStr = retStr + v
+	}
+
+	if len(item.ProviderAddresses) >= 0 {
+		retStr = retStr + "&"
+		retStr = retStr + constant.OVERRIDE_PROVIDERS_KEY
+		retStr = retStr + "="
+		retStr = retStr + strings.Join(item.ProviderAddresses, ",")
+	}
+
+	return retStr, nil
+}
+func getEnabledString(item ConfigItem, config ConfiguratorConfig) string {
+	retStr := "&enabled="
+	if len(item.Type) == 0 || item.Type == GeneralType {
+		retStr = retStr + strconv.FormatBool(config.Enabled)
+	} else {
+		retStr = retStr + strconv.FormatBool(item.Enabled)
+	}
+	return retStr
+}
+
+const (
+	ScopeApplication = "application"
+	GeneralType      = "general"
+)
+
+type ConfiguratorConfig struct {
+	ConfigVersion string       `yaml:"configVersion"`
+	Scope         string       `yaml:"scope"`
+	Key           string       `yaml:"key"`
+	Enabled       bool         `yaml:"enabled"`
+	Configs       []ConfigItem `yaml:"configs"`
+}
+type ConfigItem struct {
+	Type              string            `yaml:"type"`
+	Enabled           bool              `yaml:"enabled"`
+	Addresses         []string          `yaml:"addresses"`
+	ProviderAddresses []string          `yaml:"providerAddresses"`
+	Services          []string          `yaml:"services"`
+	Applications      []string          `yaml:"applications"`
+	Parameters        map[string]string `yaml:"parameters"`
+	Side              string            `yaml:"side"`
+}
```

Comment:
hey, guy, this is not good. u should put the const/type in the header of a file.